mteb/encoder_interface.py

```diff
@@ -63,6 +63,24 @@ def encode(
         """
         ...
 
+    def load_task_sample(
+        self,
+        sentences: Sequence[str],
+        task_name: str,
+        prompt_type: PromptType | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Load a sample of the task to be used for encoding. This is used for cde models.
+
+        Args:
+            sentences: The sentences to encode.
+            task_name: The name of the task. Sentence-transformers uses this to
+                determine which prompt to use from a specified dictionary.
+            prompt_type: The name type of prompt. (query or passage)
+            kwargs: Additional arguments to pass to the encoder.
+        """
+        ...
```

Comment:
I don't think this is needed to do what you want, so to keep the implementation simple and to avoid breaking backward compatibility I would prefer if we didn't do this